tests/Datagrid/PagerTest.php

```diff
@@ -291,19 +311,14 @@ public function testGetLinks(): void
         $this->assertSame([45, 46, 47, 48, 49, 50], $this->pager->getLinks());
     }
 
-    /**
-     * NEXT_MAJOR: remove legacy group.
-     *
-     * @group legacy
-     */
     public function testHaveToPaginate(): void
     {
         $this->assertFalse($this->pager->haveToPaginate());
 
         $this->pager->setMaxPerPage(10);
         $this->assertFalse($this->pager->haveToPaginate());
 
-        $this->setProperty($this->pager, 'nbResults', 100);
+        $this->countResults = 100;
```

Comment:
```
$this->pager->expects($this->any())
            ->method('countResults')
            ->willReturn(100);
```

this way you don't need the countResult property, no ?

Reply:
Indeed :+1: I thought I would use it in multiple places. But actually its only one place